common/src/sql/dbinit.sql

```diff
@@ -127,13 +130,21 @@ CREATE TABLE omicron.public.service (
     sled_id UUID NOT NULL,
     /* The IP address of the service. */
     ip INET NOT NULL,
+    /* The UDP or TCP port on which the service listens. */
+    port INT4 CHECK (port BETWEEN 0 AND 65535) NOT NULL,
```

Comment:
This is *probably* low priority, but should we also store the protocol? We're now storing services that aren't just TCP.

(Totally okay to defer)

Reply:
That's a fair question.  SRV records don't have that (I think).  I don't think it's necessary in the same way that the IP and port are.  I think it's assumed clients know what kind of thing they're going to talk to, just not how to reach it.

Reply:
The protocol would be part of the record if I'm reading [this](https://en.wikipedia.org/wiki/SRV_record#Record_format) right:

>A SRV record has the form:
>`_service._proto.name. ttl IN SRV priority weight port target.`

(`_proto` here being the transport protocol, e.g. `_tcp`, `_udp`)

But looks like it's all just hard-coded to TCP right now:

https://github.com/oxidecomputer/omicron/blob/5d39727039cca86a7fef48d35f3675187386c333/internal-dns/src/names.rs#L53-L76

Reply:
Thank you -- I'd forgotten about that!